src/VisualStudio/Core/Def/Implementation/DebuggerIntelliSense/DebuggerTextView.cs

```diff
@@ -60,6 +64,25 @@ public bool IsImmediateWindow
             get;
         }
 
+        public uint StartBufferUpdate()
+        {
+            // null in unit tests
+            if (_debuggerTextLinesOpt == null)
+            {
+                return 0;
+            }
+
+            _debuggerTextLinesOpt.GetStateFlags(out var bufferFlags);
+            _debuggerTextLinesOpt.SetStateFlags((uint)((BUFFERSTATEFLAGS)bufferFlags & ~BUFFERSTATEFLAGS.BSF_USER_READONLY));
```

Comment:
We also have this same stuff going on in DebuggerIntellisenseFilter.cs:

https://github.com/dotnet/roslyn/blob/70c4eaf78e1962a34f051dda2d5f2ab6c789af20/src/VisualStudio/Core/Def/Implementation/DebuggerIntelliSense/DebuggerIntellisenseFilter.cs#L98-L99

Was this the right place for that all along? It seems like yes, because commits indeed can come through not through the commanding system. Should that other code be deleted?

Reply:
I think the other code needs to stay there so that characters are written to to buffer as you type.

Reply:
@AmadeusW Do we have this happening in our new system as well?